canvas/gradient.go

```diff
@@ -0,0 +1,190 @@
+package canvas
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"math"
+
+	"fyne.io/fyne"
+)
+
+// Gradient describes a gradient between two colors
+type Gradient struct {
+	baseObject
+
+	Generator func(w, h int) image.Image
```

Comment:
For raster it made sense for this to be public, here it does not

Reply:
imho it still makes sense for the Generator to be public as you can combine multiple gradients into an image or interact with the image before sending it to the driver via an Image object, etc